opendevin/action/base.py

```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass
+
+from ..controller import AgentController
+
+@dataclass
+class Action:
+    pass
+
+    def run(self, controller: AgentController) -> str:
+        raise NotImplementedError
+
+    def to_dict(self):
+        return {
+            "action_type": self.__class__.__name__,
+            "args": self.__dict__
```

Comment:
I recently added a `message` field for all Actions (Events in the old system). Might be worth keeping

Reply:
Did you merge this one already though?

Reply:
Not sure if I had at the time, but it's in now 😄 

https://github.com/OpenDevin/OpenDevin/blob/4aa24eb41d51392b17bb4a2df50554fd73f89834/opendevin/lib/event.py#L7